src/ca_biositing/pipeline/ca_biositing/pipeline/utils/interactive_commodity_mapper.py

```
@@ -0,0 +1,1313 @@
+                    else:
+                        print("Invalid format. Please enter exactly 8 digits.")
+                        continue
+                break  # Exit main choice loop after successful manual entry
+
+            if choice == 'u':
+                if approved:
+                    last_match = approved.pop()  # Remove last match
+                    # Put the resource back at the beginning of pending for re-review
+                    pending.insert(i, {
+                        'resource': last_match['resource'],
+                        'candidates': find_best_matches(last_match['resource']['name'], load_usda_commodities())
```

Ruff (0.15.0)

[error] Undefined name `load_usda_commodities`

(F821)